homeassistant/components/schedule/__init__.py

```diff
@@ -295,7 +295,10 @@ def _update(self, _: datetime | None = None) -> None:
             (
```

Comment:
We should rewrite this `next` statement to a

```python
for time_range in todays_scheule:
    if …:
        self._attr_state = 
        break

else:
    self._attr_state = STATE_OFF
```

And then add some comments.

Reply:
Done.